app/controllers/records_controller.rb

```diff
@@ -0,0 +1,26 @@
+class RecordsController < ApplicationController
+  def index
+    @category = current_category
+  end
+
+  def new; end
+
+  def create
+    @record = current_user.records.new(record_params)
+    if @record.save
+      redirect_to category_records_path(current_category), notice: 'You completed this transaction successfully'
+    else
+      render :new, alert: 'New transaction could not be created.'
+    end
+  end
+
+  def current_category
+    @categories = current_user.categories.includes(:records).find(params[:category_id])
+  end
+
+  private
+
+  def record_params
+    params.require(:record).permit(:name, :amount, category_ids: [])
+  end
+end
```

Comment:
- [ ] I am able to visit different **Record** routes (although it throws errors). But this should not be so. I would recommend you add the `before_action :authenticate_user!` helper at the top of the controller to prevent this.